ccm_web/server/src/lti/lti.service.ts

```diff
@@ -45,7 +47,31 @@ export class LTIService implements BeforeApplicationShutdown {
 
     // Redirect to the application root after a successful launch
     provider.onConnect(async (token: IdToken, req: Request, res: Response) => {
-      logger.debug(JSON.stringify(token.userInfo))
+      logger.debug(`LTI launch is successfull! ${JSON.stringify(token.userInfo)}`)
+      const customLTIVariables = token.platformContext.custom
+      if (customLTIVariables == null) {
+        return res.json({ lti_error: 'LTI launch is missing custom attributes, please add it in LTI configutation in Canvas' })
+      }
+      const loginId = customLTIVariables.login_id as string
+      const [record, created] = await this.userService.upsertUser({
+        firstName: token.userInfo.given_name,
+        lastName: token.userInfo.family_name,
+        email: token.userInfo.email,
+        loginId: loginId,
+        ltiId: token.user
+      })
+      /* exclusively treating null value as false, But in really the const `created` won't be null with mysql dialect.
```

Comment:
This comment could use some clarification. It kinda seems like `created` would only return `false` if the object wasn't changed at all? Otherwise, looking at the docs for Sequelize, it makes it sound like it would return `true` if it was created or updated. And if it failed for some reason, then an error would be thrown, meaning you wouldn't have access to `record` or `created`. This might need some finessing/research.

Reply:
Well, I saw `created` in the logs on the first launch, and `updated` on a subsequent, so maybe I don't quite understand how this works yet. I'll keep looking at it.

Reply:
Okay, it makes sense to me now that `created` would be `true` if it's brand new and `false` otherwise (at least for MySQL). So I guess the `created ?? false` logic means that if we're not using MySQL, it would always be `updated`, which seems reasonable (and is an edge case anyway).

Reply:
Cool, thanks, resolving...